server.js

```diff
@@ -0,0 +1,82 @@
+var _ = require("lodash"),
+  bodyparser = require("body-parser"),
+  express = require("express"),
+  jsdom = require("jsdom"),
+  request = require("request"),
+  geocoderCn = require("./lib/geocoder.js"),
+  makeObjAndStringify = require("./lib/makeobjandstringify.js"),
+  piglatinify = require("./lib/piglatinify.js"),
+  randomJoke = require("./lib/jokes.js"),
+  randomString = require("./lib/strings.js");
+var app = express();
+var port = process.env.PORT || 3000;
+
+// Limit portions of app bringing in, so it doesn't load EVERYTHING
+app.use(bodyparser.json()); // Use the JSON portion of body parser
+app.use(bodyparser.urlencoded({ extended: true })); // Use url decoder
+app.use(express.static(__dirname + "/app/"));  // Links Our App Folder
+
+////////////////////// Endpoints/Routes //////////////////////////
+app.get("/", function(req, res) {
+  res.sendFile("index.html");     // We can send files from app folder
+  //res.send("hello, universe");
+});
+
+app.get("/jokes", function(req, res) {
+  res.json(randomJoke());
+});
+
+app.get("/strings", function(req, res) {
+  res.send(randomString());
+});
+
+// TODO: Hit Remote Server ONLY if haven't already loaded value before.
+app.get("/npmname", function(req, res) {
+  var siteAddress = "https://www.npmjs.com/package/wolverine";
+  request({ uri: siteAddress }, function(err, response, body) {
+    // Returns a JSON DOM Object; parses for the proper text
+    jsdom.env( {
+      html: (response["req"]["res"]["body"]),
+      scripts: [ "http://code.jquery.com/jquery-1.6.min.js" ],
+      done: function(err, window) {
```

Comment:
this manages your async here; once the response comes back and is parsed, the stuff in this done block will execute in order (not async), because you're not calling any async functions here.